lib/compress/zstd_double_fast.c

```diff
@@ -198,6 +198,9 @@ size_t ZSTD_compressBlock_doubleFast_generic(
         }   }
 
         ip += ((ip-anchor) >> kSearchStrength) + 1;
+#if defined(__aarch64__)
+        PREFETCH_L1(ip+256);
+#endif
```

Comment:
For X86 too?

Reply:
> For X86 too?
There is gains on X86 too. But It may negative effects on decompression.

	x86				compression		decompression
	clang9.0			1.13%			1.54%
	gcc9.2.0			3.26%			-1.55%

Reply:
Don't worry about the decompression speed for compression-only changes (unless you are changing how the data gets compressed and don't have identical compressed output). That is just noise, and we shouldn't take it into account.

Reply:
I've found on x86 it is better to put this prefetch before line 142 (hash table update). Does that work on aarch64 too?

Reply:
> I've found on x86 it is better to put this prefetch before line 142 (hash table update). Does that work on aarch64 too?

Putting this prefetch before line 142 is worse in arrch64/clang9.0.0，is similar in arrch64/gcc9.2.0.

Reply:
@terrelln  Do I need to remove the __aarch64__ switch?

Reply:
Let's land this patch with the `#if defined(__aarch64__)` then put up a second patch that just deletes them. I don't want to block clear aarch64 improvements with x86 benchmarking.